### PR TITLE
Uxd 339

### PR DESCRIPTION
**Pull request by ThyeeZz**

### CODE REVIEW

File: src/components/textSelection/CommentDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        ></textarea>
+      </div>
+      <div className="comment-footer">
+        <span
+          tabIndex={0}
+          role='button'
+          className={`${!message.length ? 'disabled' : ''} button`}
+          onClick={handleSendComment}
+          onKeyDown={handleSendComment}
+        >Send</span>
```
Comment (Tumao727) [inline src/components/textSelection/CommentDialog.jsx:50]: i18n

File: src/components/textSelection/TextSelectionMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const TextSelectionMenu = ({ options, showDialog }) => {
+  const [style, setStyle] = useState({
+    visibility: 'hidden',
+    zIndex: -100,
+  });
+
+
+  useEffect(() => {
+    setStyle(options)
+  }, [options]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (nameczz) [inline src/components/textSelection/TextSelectionMenu.jsx:15]: why need eslint-disable-line react-hooks/exhaustive-deps?

File: src/components/textSelection/CommentDialog.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        <span
+          tabIndex={0}
+          role='button'
+          className='button cancel-btn'
+          onClick={handleHideDialog}
+          onKeyDown={handleHideDialog}
+        >Cancle</span>
```
Comment (Tumao727) [inline src/components/textSelection/CommentDialog.jsx:57]: typo and i18n

File: src/components/textSelection/TextSelectionMenu.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          className="content-item" 
+          role='button'
+          tabIndex={-1} 
+          onClick={showDialog} 
+          onKeyDown={showDialog}
+        >
+          <p className="icon-wrapper">
+            <img src={comment} alt="comment-logo" />
+          </p>
+          <span>Comment</span>
```
Comment (Tumao727) [inline src/components/textSelection/TextSelectionMenu.jsx:30]: i18n

File: src/components/textSelection/TextSelectionMenu.jsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        </div>
+        |
+        <a href="http://www.baidu.com" className="content-item">
```
Comment (Tumao727) [inline src/components/textSelection/TextSelectionMenu.jsx:33]: why github label use baidu's url?

File: src/hooks/index.jsx [hunk 6]
```diff
@@ -14,3 +15,58 @@ export const useMobileScreen = () => {
...
   return screenWidth;
 };
+
+export const useSelectMenu = (setOptions) => {
+  // titlebar height:60; self height: 26; margin 8
+  const offsetTop = 60 + 36 + 8;
+  const docWrapper = document.getElementsByClassName('doc-wrapper')[0];
+  let docWrapWidth = 0;
+  if (docWrapper) {
+    docWrapWidth = parseInt(window.getComputedStyle(docWrapper).width);
```
Comment (Tumao727) [inline src/hooks/index.jsx:25]: why not use docWrapper.offsetWidth?

File: src/hooks/index.jsx [hunk 7]
```diff
@@ -14,3 +15,58 @@ export const useMobileScreen = () => {
...
+  }
+  const { innerWidth } = window;
+  // left blank width plus menu-bar width
+  const offsetLeft = (innerWidth - docWrapWidth) / 2 + 250;
+
+  // max width of single line
+  const maxLineWidth = docWrapWidth - 280 - 250 - 64 - 8;
```
Comment (nameczz) [inline src/hooks/index.jsx:32]: explain the magic number.

File: src/components/textSelection/TextSelectionMenu.jsx (hunk 2 above)
Comment (ThyeeZz) [reply src/components/textSelection/TextSelectionMenu.jsx:15]: cause eslint keeps prompting me to add another dependency that I don’t need

**Review by nameczz — APPROVED** [review]

**Pull request merged**